native/test/NativeCaller.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./BaseNativeInterface.sol";
+
+contract NativeCaller {
+
+    BaseNativeInterface nativeContract = BaseNativeInterface(0x00000000000000000000000000000000DeaDBeef);
+
+    //This function tests that calling a readonly method on a Native Smart Contract using staticcall works
+    // and that calling then a readwrite method works again, so the statedb is readwrite again.
+    function testStaticCallOnReadonlyMethod() public returns (uint32) {
+	address contractAddr = address(nativeContract);
+	(bool success, bytes memory result) = contractAddr.staticcall{gas:10000}(
+            abi.encodeWithSignature("retrieve()")
+        );
+	(uint32 a) = abi.decode(result, (uint32));
+	//Check that statedb is readwrite again
+	(bool success2, bytes memory result2) = contractAddr.call{gas:25000}(
```

Comment:
the gas value can be encapsulated in contract costants, something like this:
```
uint256 private constant staticCallGasLimit = 10000;
uint256 private constant callGasLimit = 25000;
```

Reply:
Done